proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -140,6 +140,7 @@
     @Getter
     @Setter
     private ForgeServerHandler forgeServerHandler;
+    private boolean disconnecting;
```

Comment:
The disconnect method currently only closes the channel after 500ms; without this flag, this method can be called multiple times within a 500ms time-frame and still log in the console that the user has disconnected and create a runnable to close the channel and send a kick packet.

By setting this as a dirty flag, we can ignore additional cases.